### PR TITLE
Modernize Phase2TrackerDigiProducer

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- changed to a global module
- improved memory handling
- avoid unnecessary recalculations

#### PR validation:

Code compiles.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@makortel FYI

File: EventFilter/Phase2TrackerRawToDigi/plugins/Phase2TrackerDigiProducer.cc [hunk 1]
```diff
@@ -193,46 +204,40 @@ namespace Phase2Tracker {
...
-          it = it2;
-        }
+      }
+    }
 
-        edm::DetSetVector<Phase2TrackerDigi> proc_raw_dsv(sorted_and_merged, true);
-        pr->swap(proc_raw_dsv);
-        event.put(std::unique_ptr<edm::DetSetVector<Phase2TrackerDigi>>(pr), "ProcessedRaw");
-        delete buffer;
+    // store digis in edm collections
```
Comment (Dr15Jones) [inline EventFilter/Phase2TrackerRawToDigi/plugins/Phase2TrackerDigiProducer.cc:220]: Note, this was moved out of the outer most `for` loop. This avoids redoing the work again and again only to replace what was being pushed into the `edm::Event`.

**Comment by Dr15Jones** [comment]
This was found as part of our review of `stream::` modules using the Run and/or LuminosityBlock transitions.

**Comment by Dr15Jones** [comment]
please test

**Comment by mandrenguyen** [comment]
+reconstruction
No changes to reco comparisons

**Comment by mmusich** [comment]
>  No changes to reco comparisons

just for the record, that's expected because this module is not used in any reconstruction sequence, nor tested in any unit tests (AFAIU).
I am wondering what prompted the changes proposed here (not that I am against them).

**Comment by smorovic** [comment]
+daq
global module does look like a good fit.

**Comment by Dr15Jones** [comment]
> I am wondering what prompted the changes proposed here (not that I am against them).

We are in the process of modifying the framework to require stream modules to explicitly request to handle Run and/or LuminosityBlock transitions. We are in the process of modifying the ~200 modules which need to be changed. As part of that change, we sometimes just move the work done in Run or LuminosityBlock calls to the Event call. This change started as one of those changes but I felt additional changes would improve things further.

**Comment by rappoccio** [comment]
+1

**Pull request merged**